ControlRoom/Controllers/UIState.swift

```diff
@@ -0,0 +1,17 @@
+//
+//  UIState.swift
+//  ControlRoom
+//
+//  Created by Dave DeLong on 2/16/20.
+//  Copyright © 2020 Paul Hudson. All rights reserved.
+//
+
+import Combine
+
+class UIState: ObservableObject {
+    static let shared = UIState()
```

Comment:
It's not quite clear to me why `UIState` is a singleton.
Since it's passed to the `MainView` as `environmentObject` we could just store a reference in `MainWindowController` ad use it.
Am I missing something?

[Check out this patch](https://github.com/twostraws/ControlRoom/files/4211071/0001-Remove-UIState-singleton.patch.zip)

Reply:
The idea of making it a singleton is that there's only a single preferences window, a single "create simulator" sheet, a single about window, etc, so there should only be a single object to manage whether or not they're visible.

Reply:
Making it an `environmentObject` is so views can conveniently bind to it.

Reply:
Mh you didn't quite convinced me.
Singletons are not very functional and in this case I can't see the benefit of using it since we can just create an instance of `UIState` in `MainViewController` (like it has been done with `Preferences`) and everything will be forwarded to the sub-views through `environmentObject`.

Reply:
@MarioIannotta a single app only has a single state. It would not be _correct_ to have multiple instances of `UIState`, nor even to _allow_ multiple instances of it.